lib/foreman_tasks/tasks/export_tasks.rake

```diff
@@ -317,7 +323,7 @@ namespace :foreman_tasks do
     format = ENV['TASK_FORMAT'] || 'html'
     export_filename = ENV['TASK_FILE'] || generate_filename(format)
 
-    tasks = ForemanTasks::Task.search_for(filter).order(:started_at => :desc).with_duration.distinct
+    tasks = ForemanTasks::Task.search_for(filter).with_duration.order(:started_at => :desc)
```

Comment:
If I recally correctly, certain search can cause tasks to be duplicated by using multiple left joins. We cannot really just drop `distinct`. Of course we can replace it with a different mechanism (`group by` should work well in this context),  but we cannot just drop it

Reply:
But how should I use `group by`? Because I found out, that it only works when I group by each column, which is very slowly.

Reply:
Something like this could work
```
From 24537a495f44f49281bad45647a9229a783061e1 Mon Sep 17 00:00:00 2001
From: Adam Ruzicka <aruzicka@redhat.com>
Date: Wed, 16 Feb 2022 13:30:01 +0100
Subject: [PATCH] ...

---
 lib/foreman_tasks/tasks/export_tasks.rake | 20 +++++++++++---------
 1 file changed, 11 insertions(+), 9 deletions(-)

diff --git lib/foreman_tasks/tasks/export_tasks.rake lib/foreman_tasks/tasks/export_tasks.rake
index a48483f..76553c4 100644
--- lib/foreman_tasks/tasks/export_tasks.rake
+++ lib/foreman_tasks/tasks/export_tasks.rake
@@ -255,19 +255,20 @@ namespace :foreman_tasks do
       end
     end
 
-    def html_export(workdir, tasks)
+    def html_export(workdir, tasks, task_scope)
       PageHelper.copy_assets(workdir)
+      ids = tasks.pluck(:id)
 
       renderer = TaskRender.new
       count = 0
-      total = tasks.count(:all)
+      total = ids.count
       index = File.open(File.join(workdir, 'index.html'), 'w')
 
       File.open(File.join(workdir, 'index.html'), 'w') do |index|
         PageHelper.pagify(index) do |io|
           PageHelper.generate_with_index(io) do |index|
-            tasks.pluck(:id).in_groups_of(1000, false).each do |ids|
-              tasks.where(id: ids).each do |task|
+            ids.in_groups_of(1000, false).each do |ids|
+              task_scope.where(id: ids).each do |task|
                 content = with_error_handling(task) { renderer.render_task(task) }
                 if content
                   File.open(File.join(workdir, "#{task.id}.html"), 'w') { |file| PageHelper.pagify(file, content) }
@@ -323,21 +324,22 @@ namespace :foreman_tasks do
     format = ENV['TASK_FORMAT'] || 'html'
     export_filename = ENV['TASK_FILE'] || generate_filename(format)
 
-    tasks = ForemanTasks::Task.search_for(filter).with_duration.order(:started_at => :desc)
+    id_scope = ForemanTasks::Task.search_for(filter).order(:started_at => :desc).group(:id, :started_at)
+    task_scope = ForemanTasks::Task.search_for(filter).with_duration
 
     puts _("Exporting all tasks matching filter #{filter}")
-    puts _("Gathering #{tasks.count(:all)} tasks.")
+    puts _("Gathering #{id_scope.count(:all).count} tasks.")
     case format
     when 'html'
       Dir.mktmpdir('task-export') do |tmp_dir|
-        html_export(tmp_dir, tasks)
+        html_export(tmp_dir, id_scope, task_scope)
         system("tar", "czf", export_filename, tmp_dir)
       end
     when 'html-dir'
       FileUtils.mkdir_p(export_filename)
-      html_export(export_filename, tasks)
+      html_export(export_filename, id_scope)
     when 'csv'
-      csv_export(export_filename, tasks)
+      csv_export(export_filename, id_scope)
     else
       raise "Unkonwn export format '#{format}'"
     end
-- 
2.31.1
```